server/manager/src/main/java/org/apache/accumulo/manager/Manager.java

```diff
@@ -225,6 +226,35 @@ public boolean stillManager() {
     return getManagerState() != ManagerState.STOP;
   }
 
+  Fate<Manager> fate() {
+    Fate<Manager> fate = fateReady.get();
+    if (fate != null) {
+      // it's ready, just return it
+      return fate;
+    }
+
+    // it's not ready yet
+    long retryTime = 500; // millis
+
+    // create informative warning
+    String msgPrefix = "Unexpected use of fate in thread " + Thread.currentThread().getName()
+        + " at time " + System.currentTimeMillis();
+    log.warn("{} blocked until fate starts", msgPrefix,
+        new IllegalStateException("Attempted fate action before fate was started; "
+            + "if this doesn't make progress, please report it as a bug to the developers"));
```

Comment:
We could reduce this to debug if this is a normal situation, and not something to warn the users about. I'm just not sure.

Reply:
@dlmarion Any thoughts on reducing this to debug?

Reply:
Could just log the warning. I am not sure how useful the full stack trace is.

Reply:
I was hoping that the stack trace would help us debug where it was coming from, if it was something that shouldn't be doing whatever it was doing.

Reply:
So, regarding the logging, you could do something like this instead:

```
  if (!fateReadyLatch.await(1, MINUTE)) {
    // LOG HERE
  }
```

The waiting should be short-circuited when the latch count goes to zero.

Reply:
I would suggest making the wait time large so that you don't have to put it into a loop. If the wait time has expired and the await() method returns false, then you could shut down the manager (if the wait time is like 10 mins or something ridiculous)

Reply:
I'm not sure about shutting down the manager. The manager might not be able to help that a tserver or client is asking it to do something when it's not ready. But, I do like the idea of logging periodically if it isn't making progress. I'd rather put it into a loop for that. I could make it 5 minutes and log a warning. I used trace before when I was looping 250ms, but warning I think is most appropriate if it's stuck for 5 minutes.

Reply:
I'll make it so the first warning doesn't occur until at least 30 seconds of waiting. And subsequent warnings every 5 minutes that progress isn't made.